### PR TITLE
Fix/gh r musl

### LINKED ISSUE #539: [bug]: gh-r linux wrong arch binary -- downloads aarch64 instead of x86_64
### What happened?

On my x86_64 system running **Kubuntu 23.04**, `znit` `gh-r` downloads the wrong arch version for [navi](https://github.com/denisidoro/navi). It downloads `navi-v2.22.1-aarch64-unknown-linux-gnu.tar.gz` instead of `navi-v2.22.1-x86_64-unknown-linux-musl.tar.gz`.

Similar bug report: #374 

### Steps to reproduce

I'm using the `binary-symlink` annex to load [navi](https://github.com/denisidoro/navi). It downloads the correct arch for all the other bins I load. Here's the snippet from my `.zshrc`:

```
zi lucid from"gh-r" lbin'!' nocompile atpull'%atclone' for \
    pemistahl/grex \
    denisidoro/navi \
  atclone'mkdir -p $ZPFX/{bin,man/man1} ;
      cp -vf fd*/fd.1 $ZINIT[MAN_DIR]/man1/' \
    @sharkdp/fd \
  atclone'mkdir -p $ZPFX/{bin,man/man1} ;
      cp -vf lsd*/lsd.1 $ZINIT[MAN_DIR]/man1/' \
    lsd-rs/lsd \
  atclone'mkdir -p $ZPFX/{bin,man/man1,man/man5} ;
      cp -vf completions/exa.zsh completions/_exa ;
      cp -vf man/exa.1 $ZINIT[MAN_DIR]/man1/ ;
      cp -vf man/exa_colors.5 $ZINIT[MAN_DIR]/man5/' \
    ogham/exa \
  atclone'mkdir -p $ZPFX/{bin,man/man1}' \
  dl"$GH_RAW_URL/junegunn/fzf/master/shell/completion.zsh -> _fzf_completion ;
     $GH_RAW_URL/junegunn/fzf/master/man/man1/fzf.1 -> $ZINIT[MAN_DIR]/man1/fzf.1" \
    junegunn/fzf \
  atclone'mkdir -p $ZPFX/{bin,man/man1} ;
      cp -vf ripgrep*/doc/rg.1 $ZINIT[MAN_DIR]/man1/' \
  lbin'!ripgrep*/rg' \
    BurntSushi/ripgrep \
  atclone'mv cheat-linux-amd64 cheat' \
  dl"$GH_RAW_URL/cheat/cheat/master/scripts/cheat.zsh -> _cheat_completion" \
    cheat/cheat \
  atclone'mkdir -p $ZPFX/{bin,man/man1} ;
      cp -vf micro*/micro.1 $ZINIT[MAN_DIR]/man1/' \
    zyedidia/micro
```

### Relevant output
Here's the `zinit` output when installing:
```
Downloading denisidoro/navi…
(Requesting `navi-v2.22.1-aarch64-unknown-linux-gnu.tar.gz'…)
######################################################################## 100.0%                                                               
[ziextract] Unpacking the files from: `navi-v2.22.1-aarch64-unknown-linux-gnu.tar.gz'…
[ziextract] Successfully extracted and assigned +x chmod to the file: navi.
==> linkbin annex: Created navi soft link & set +x on the navi binary
```

Here's the `zsh` output when trying to execute the `aarch64` version of `navi`:
```
$ navi
zsh: exec format error: navi
```

### Operating System & Version

OS: linux-gnu | Vendor: ubuntu | Machine: x86_64 | CPU: x86_64 | Processor: x86_64 | Hardware: x86_64

### Zsh version

zsh 5.9 (x86_64-ubuntu-linux-gnu)

### Terminal emulator

xterm-256color (tilix)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by vladdoster**
## Related Issue(s)

Closes #539

## Types of changes 

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation change
- [ ] New feature (non-breaking change which adds functionality)

## Checklist:

- [X] All new and existing tests passed.
- [ ] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: zinit-install.zsh [hunk 1]
```diff
@@ -554,8 +554,8 @@ builtin source "${ZINIT[BIN_DIR]}/zinit-side.zsh" || {
     local c cfile bkpfile
     # The plugin == . is a semi-hack/trick to handle 'creinstall .' properly
     [[ $user == % || ( -z $user && $plugin == . ) ]] && \
-        completions=( "${plugin}"/**/_[^_.]*~*(*.zwc|*.html|*.txt|*.png|*.jpg|*.jpeg|*.js|*.md|*.yml|*.yaml|*.py|*.ri|_zsh_highlight*|/zsdoc/*|*.ps1)(DN^/) ) || \
-        completions=( "${ZINIT[PLUGINS_DIR]}/${id_as//\//---}"/**/_[^_.]*~*(*.zwc|*.html|*.txt|*.png|*.jpg|*.jpeg|*.js|*.md|*.yml|*.yaml|*.py|*.ri|_zsh_highlight*|/zsdoc/*|*.ps1)(DN^/) )
+        completions=( "${plugin}"/**/_[^_.]*~*(*.zwc|*.html|*.txt|*.png|*.jpg|*.jpeg|*.js|*.md|*.yml|*.ri|_zsh_highlight*|/zsdoc/*|*.ps1)(DN^/) ) || \
+        completions=( "${ZINIT[PLUGINS_DIR]}/${id_as//\//---}"/**/_[^_.]*~*(*.zwc|*.html|*.txt|*.png|*.jpg|*.jpeg|*.js|*.md|*.yml|*.ri|_zsh_highlight*|/zsdoc/*|*.ps1)(DN^/) )
```
Comment (pschmitt) [inline zinit-install.zsh:558]: Please rebase. This is from #528 and should stay in!

File: .github/workflows/documentation.yaml [hunk 2]
```diff
@@ -18,21 +18,20 @@ jobs:
...
+        run: |
+          echo "### Documentation summary" >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY # this is a blank line
+          if ! git --no-pager diff --exit-code; then
+            echo "::error:: Z-shelldocs are out-of-date." >> $GITHUB_STEP_SUMMARY
+            echo 'To regenerate, run `make doc`' >> $GITHUB_STEP_SUMMARY
+            exit 1
+          else
+            echo "::notice:: Z-shelldocs are up-to-date." >> $GITHUB_STEP_SUMMARY
+          fi
```
Comment (pschmitt) [inline .github/workflows/documentation.yaml:37]: Please rebase. This reverts a bunch of PR that I merged in. This here is #535

**Review by pschmitt — CHANGES_REQUESTED** [review]
Please rebase.

**Comment by vladdoster** [comment]
Closed in favor of #564.

**Pull request closed without merging**